src/cpp/fastdds/domain/DomainParticipantFactory.cpp

```diff
@@ -424,6 +438,7 @@ void DomainParticipantFactory::reset_default_participant_qos()
         eprosima::fastrtps::ParticipantAttributes attr;
         XMLProfileManager::getDefaultParticipantAttributes(attr);
         utils::set_qos_from_attributes(default_participant_qos_, attr.rtps);
+        default_domain_id_ = attr.domainId;
```

Comment:
As far as I can tell, if the only change in the profile is the domain ID, then this will not take effect. Is that what we want?

Reply:
It was not being applied, you were right. With the changes included in [8045a14](https://github.com/eProsima/Fast-DDS/pull/4534/commits/8045a140dbdb3e038df1b9b1e99aeee4328e286e), I've tested your scenario successfully:

No setting the XML file takes default domain 0
![image](https://github.com/eProsima/Fast-DDS/assets/120394830/ad7a7db9-9843-4633-8d7b-b4cbd9ecb1a6)


Setting the specified XML file takes domain 1
![image](https://github.com/eProsima/Fast-DDS/assets/120394830/25e89f32-b220-4f7c-9045-170b9e669a11)